src/cennznut/v0/mod.rs

```diff
@@ -99,30 +133,139 @@ impl Decode for CENNZnutV0 {
     }
 }
 
-impl Validate<ModuleDomain> for CENNZnutV0 {
-    /// Validates a CENNZnut by (1) looking for `module_name` and `method_name` and (2) executing the
-    /// Pact interpreter if constraints exist
-    fn validate(
+impl CENNZnutV0 {
```

Comment:
We could implement the two traits ValidateContractCall and ValidateRuntimeCall that I mentioned in my other comment for both CENNZnutV0 and CENNZnut. In the latter case, CENNZnut would check the version and call the same interface from the corresponding version of CENNZnutV0.